logdetective/server/emoji.py

```diff
@@ -0,0 +1,83 @@
+import datetime
+import asyncio
+
+from typing import List
+from collections import Counter
+
+import gitlab
+
+from logdetective.server.models import TimePeriod
+from logdetective.server.database.models import (
+    Comments,
+    Reactions,
+    GitlabMergeRequestJobs,
+)
+
+
+async def collect_emojis(gitlab_conn: gitlab.Gitlab, period: TimePeriod):
+    """
+    Collect emoji feedback from logdetective comments saved in database.
+    Check only comments created in the last given period of time.
+    """
+    comments = Comments.get_since(period.get_period_start_time())
+    comments_for_gitlab_connection = [
+        comment for comment in comments if comment.forge == gitlab_conn.url
+    ]
+    await collect_emojis_in_comments(comments_for_gitlab_connection, gitlab_conn)
+
+
+async def collect_emojis_for_mr(
+    project_id: int, mr_iid: int, gitlab_conn: gitlab.Gitlab, period: TimePeriod
+):
+    """
+    Collect emoji feedback from logdetective comments in the specified MR.
+    Check only MR comments created in the last given period of time.
+    """
+    mr_jobs = GitlabMergeRequestJobs.get_by_mr_iid(gitlab_conn.url, project_id, mr_iid)
+    comments_for_mr = [Comments.get_by_mr_job(mr_job) for mr_job in mr_jobs]
+    comments = [
+        comment
+        for comment in comments_for_mr
+        if comment.created_at.replace(tzinfo=datetime.timezone.utc)
+        > period.get_period_start_time()  # noqa: W503 ruff vs flake
```

Comment:
Can you remove the TimePeriod argument here (or make it optional, if you think it's still useful)?

I think if we're making a request on a _specific_ merge request, we should look up all of its comments, even if they're very old. It absolutely makes sense not to update the really old ones when we're iterating through the full set, but if we've been asked to update one specific MR, I think we should check them all.

Reply:
We're also missing the Forge argument here, which right now is not _strictly_ needed because we have the connection object, but I'm working on a patch to fix that, because we are accidentally relying on a single Gitlab connection (to one configured Gitlab instance) rather than handling things appropriately for whichever instance has triggered the message. See LD-91 in Jira.